kernel/microvm-kernel-config-aarch64

```diff
@@ -0,0 +1,2810 @@
+#
+# Automatically generated file; DO NOT EDIT.
+# Linux/arm64 4.14.256-209.484.amzn2.aarch64 Kernel Configuration
```

Comment:
Is it possible to automatically regenerate the configuration file with the new kernel version 6.6.38?

Reply:
The build does that and the config files that get put into the `result` folder will be the full config that and also calls out the new kernel. They merely act as a basis and get enhanced in the build through the build through the `olddefconfig` target.

Ideally we should spend some time to make these base configurations actual `micro-vm` configs. As they are right now they build a bunch of stuff we will probably never need inside an enclave (I saw some gpu drivers and scsi drivers scroll by when looking at the builds - there is probably a lot more room for improvement).

If you insist I can definitely update these files.

Reply:
While I agree that we should review and enhance these configurations, we should utilize the auto-generated configuration provided by the build system of the kernel we are using. The use of "old" configs could lead to confusion.

Reply:
Hi @foersleo - at Monzo we also look to build the kernel from scratch.

The way we try to do what @meerd is suggesting is by using nixpkgs' kernel config directly and patching that. See an example [here](https://github.com/monzo/aws-nitro-util/blob/master/examples/bringYourOwnKernel.nix#L21). This has lots of unnecessary stuff like you say, but it ensures we use the generated kernel config of the upstream rather than hardcoding our own.

Reply:
Hey @Cottand - thanks for pointing that out.

I have had a look at your code previously and I like it. Thanks you for diversifying the space.

I have tried out the way to build kernels as provided by nixpkgs (like you have). What I am concerned with in the way nix does their kernel builds is that they bring a common config that is tailored towards the nixos use-case. For some kernel versions they also bring a set of their own patches. I am not a big fan of that for our particular use-case as it can mask some things that go into the kernel. When we update the nixpkgs we use to build the kernel we'd have to check what has changed in their configs and or patches they add. Going the route of a fixed config we own seemed more appealing to me for the enclave use case.

I also played around with nixpkgs `linuxManualConfig` which lets you provide your own base configuration file. In that scenario I would have liked to use the `kernelPatches` to bring in the NSM driver for kernels older than 6.8 (I was trying to limit ourselves to the LTS kernels for now). For those kernels prior to v6.8 you would want to have the `CONFIG_NSM=m` setting that is coupled to the patch as part of the `extraConfig` or `extraStructuredConfig` attributes in the `kernelPatches` set. Unfortunately this seems to be broken currently (https://github.com/NixOS/nixpkgs/issues/307014) and I am not that deeply an expert in nix yet to attempt fixing that.

I thus landed at the solution you see in this PR. For continuity and not to drastic changes I used the same configs we have been using for enclave kernels in the past as a base.

I am happy to hear your thoughts on my concerns and if some of them might be completely unjustified.

---

Somewhat unrelated question regarding your linked example for my own curiosity:
* Has the native way of building kernels for nix adjusted the `text_offset` for the `Image` file on aarch64 so that these kernels do get positioned correctly and the enclave boots successfully? I admittedly have not have the time to try it and have not done the research into that kind of adjustments to the kernel images in nix (See `fixupPhase` and comment in this PR)
* Do the kernels you build through that example contain the NSM driver? As far as I can tell neither your override, nor the nix common config contain that setting. As far as I can tell the linux kernel also does not enable it by default.

Reply:
I think we have 3 options for kernel configs:

1. Use the official NixOS kernel. You gave a lot of good arguments on why today that is not necessarily the best choice for this particular kernel image we provide with nitro-cli. Customer who want to use it, definitely still can.
2. Package the exact target .config that the kernel build process emits when you build. In this case, your .config files would say they are on 6.6.
3. Package only an override of config options, similar to how Linux defconfigs work. With those, we just ingest upstream default options on future kernel version bumps - and without a mechanism to audit them we may slip functionality in that we don't want in.

I think option 2 is safest for now.

Reply:
Adjusted the config to be the outcome of the build representing a proper string for the kernel version we are building.